OpenKh.Bbs/Pmo.cs

```diff
@@ -0,0 +1,716 @@
+                            wr.Write((sbyte)(chunk.vertices[k].X * 127.0f));
+                            wr.Write((sbyte)(chunk.vertices[k].Y * 127.0f));
+                            wr.Write((sbyte)(chunk.vertices[k].Z * 127.0f));
+                            break;
+                        case CoordinateFormat.NORMALIZED_16_BITS:
+                            vertexIncreaseAmount = ((0x2 - (Convert.ToInt32(stream.Position - vertexStartPos) & 0x1)) & 0x1);
+                            for (int a = 0; a < vertexIncreaseAmount; a++)
+                                stream.Write((byte)0xAB);
+
+                            wr.Write((short)(chunk.vertices[k].X * 32767.0f));
+                            wr.Write((short)(chunk.vertices[k].Y * 32767.0f));
+                            wr.Write((short)(chunk.vertices[k].Z * 32767.0f));
+                            break;
+                        case CoordinateFormat.FLOAT_32_BITS:
+                            vertexIncreaseAmount = ((0x4 - (Convert.ToInt32(stream.Position - vertexStartPos) & 0x3)) & 0x3);
+                            for (int a = 0; a < vertexIncreaseAmount; a++)
+                                stream.Write((byte)0xAB);
+
+                            wr.Write(chunk.vertices[k].X);
+                            wr.Write(chunk.vertices[k].Y);
+                            wr.Write(chunk.vertices[k].Z);
+                            break;
+                    }
+
+                    int padding = ((int)vertexStartPos + chunk.SectionInfo.VertexSize) - (int)stream.Position;
+                    for (int p = 0; p < padding; p++)
+                        stream.Write((byte)0xAB);
+                }
+
+                // Remainder.
+                uint remainder = (uint)stream.Position % 4;
+                for (int p = 0; p < remainder; p++)
+                    stream.Write((byte)0x00);
+
+                if(j == (pmo.Meshes.Count - 1))
+                {
+                    for (uint b = 0; b < 0xC; b++)
+                        stream.Write((byte)0x00);
+
+                    for (uint b = 0; stream.Position % 0x10 != 0; b++)
+                        stream.Write((byte)0x00);
+                }
+            }
+
+            // Write textures.
+            for(int t = 0; t < pmo.texturesData.Count; t++)
+            {
+                stream.Write(pmo.texturesData[t]);
+            }
+
+            if(pmo.header.SkeletonOffset != 0)
+            {
+                BinaryMapping.WriteObject<SkeletonHeader>(stream, pmo.skeletonHeader);
+
+                for (int joint = 0; joint < pmo.jointList.Length; joint++)
+                {
+                    BinaryMapping.WriteObject<JointData>(stream, pmo.jointList[joint]);
+                }
+            }
+        }
+
+        public static bool IsValid(Stream stream) =>
+            stream.Length >= 4 &&
+            new BinaryReader(stream.SetPosition(0)).ReadUInt32() == MagicCode;
```

Comment:
Same here, you do not need `BinaryReader` as `stream.SetPosition(0).ReadUInt32()` will just work.